src/detail-page/components/charts/DevOpsChart.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { FC } from 'react';
+import { Pie } from 'react-chartjs-2';
+import { devOps } from '../../../data/charts';
+import { chartColors } from '../../../data/charts/chartColors';
+import { ChartWrapper } from './ChartWrapper';
+import { notEmpty } from '../../../shared/utils/helpers';
+
+const { data: sourceData, labelText, source, title: chartTitle, type } = devOps;
+
+const chartLabels = sourceData.map(({ label }) => label);
+const chartValues = sourceData.map(({ value }) => value);
+const chartBackgroundColors = sourceData
+  .map(({ color }) => (color ? chartColors[color].light : null))
+  .filter(notEmpty);
+const chartBorderColors = sourceData
+  .map(({ color }) => (color ? chartColors[color].dark : null))
+  .filter(notEmpty);
+
+const areColorsDefined = !sourceData.some(({ color }) => color === null);
+
+const ChartData: Chart.ChartData = {
+  labels: chartLabels,
+  datasets: [
+    {
+      data: chartValues,
+      borderWidth: 1,
+      ...(labelText && {
+        label: labelText,
+      }),
+      ...(areColorsDefined && {
```

Comment:
`areColorsDefined` - this is is check if one of our color is not missing. If there is at least one null - colors won't be added because one of their value will be null or undefined - `chart.js` allowed only string as value of color.
filtering not empty is for type guarding - because without it `chartBorderColors` has union type `(null | string)[]` & only string[] is allowed (again chart.js types).

Reply:
Gosia I understand your point :)  my question was more like - is there any situation, when we want to display our chart without colors? maybe we should just change our typings and make color requiered, not optional?